First Push of HMOS ported library

Click2cloud-Eros opened the pull request:
Add HMOS ported library.

Click2cloud-Eros committed:
First Push of HMOS ported library

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/TextPaint.java:
```diff
@@ -0,0 +1,58 @@
+
+import ohos.agp.render.Paint;
+
+public class TextPaint extends Paint {
```
Remove unused declarations in this and other classes.

Click2cloud-Eros replied:
In TextPaint class not a single method is used, So can I remove this class and directly use super class(i.e. Paint) ?

Click2cloud-Eros replied:
Remove TextPaint class.

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/RectF.java:
```diff
@@ -0,0 +1,125 @@
+/**
+ * RectF class
+ */
+public class RectF extends RectFloat {
```
So many unused declarations

Click2cloud-Eros replied:
Remove RectF class.

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/XRadarView.java:
```diff
@@ -541,38 +616,47 @@ private void drawRegionWithColor(Canvas canvas, float scale) {
         try {
-            this.getContext().getResources().openRawResource(drawable);
+            this.getContext().getResourceManager().getResource(drawable);
             drawableAvaiable = 1;
```
No use of this

Click2cloud-Eros replied:
`this.getContext().getResourceManager().getResource(drawable);`

**Use of this line:**
if this line throw exception then, drawableAvailable = 0 else drawableAvailable = 1

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/XRadarView.java:
```diff
@@ -374,9 +423,14 @@ private void drawPolygon(Canvas canvas) {
+     * @param endColor   end color
+     */
+    private void drawLayer(Canvas canvas, Color startColor, Color endColor) {
         Path path = null;
```
Null initialization is redundant here, there are similar issues at other places too, please fix them all.

Click2cloud-Eros replied:
Remove null redundancy.

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/XRadarView.java:
```diff
@@ -251,40 +262,60 @@ public XRadarView(Context context, @Nullable AttributeSet attrs, int defStyleAtt
+            int w = component.getWidth();
+            int h = component.getHeight();
+            radius = Math.min(h, w) / 2 * radarPercent;
+            MAX_TEXT_WIDTH = (int) (Math.min(h, w) / 2 * (1 - radarPercent));
```
Here there is a numeric issue - integer division in the floating-point context. Also, similar issues on several places, pls fix all.

Click2cloud-Eros replied:
solved numeric warning - integer division in the floating-point context.

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/XRadarView.java:
```diff
@@ -251,40 +262,60 @@ public XRadarView(Context context, @Nullable AttributeSet attrs, int defStyleAtt
-    }
+    public boolean onTouchEvent(Component component, TouchEvent touchEvent) {
+        HiLog.info(LABEL_LOG, "%{public}s", "onTouchEvent called");
+        switch (touchEvent.getAction()) {
```
'switch' statement has too few case labels (1), and should probably be replaced with an 'if' statement

Click2cloud-Eros replied:
Switch statement replaced by if statement.

kanaksony commented on entry/src/main/resources/base/layout/ability_main.xml:
```diff
@@ -0,0 +1,478 @@
+<?xml version="1.0" encoding="utf-8"?>
```
So many Hardcoded strings in this XML, should use `$string: ` resource

Click2cloud-Eros replied:
Move hardcoded string to string.json resource.

kanaksony commented on xradar/src/main/java/com/orzangleli/radar/XRadarView.java:
```diff
@@ -1,126 +1,136 @@
+import ohos.agp.components.Component;
+import ohos.agp.render.*;
+import ohos.agp.text.SimpleTextLayout;
+import ohos.agp.utils.*;
```
Instead, we can import explicit classes

Click2cloud-Eros replied:
Done.

kanaksony reviewed (changes requested):
So many places - proper indentation, extra and required spacing, code reformating required.

Click2cloud-Eros committed:
1) Remove RectF.java and TextPaint.java file.
2) Remove redundant code.
3) solved numeric warning - integer division in the floating-point context.
4) Switch statement replaced by if statement.
5) Move hardcoded string to string.json resource.

Click2cloud-Eros commented:
All changes Done.

Click2cloud-Eros committed:
reduce cognitive complexity of the methods and remove some code smell.

Click2cloud-Eros committed:
code cleanup using CheckStyle tool.
